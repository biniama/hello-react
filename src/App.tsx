import React, { Fragment, FC, useState } from "react";
import Title from "./Title";
import Subtitle from "./Subtitle";
import Button from "./Button";

const App: FC = () => {
  const [clickState, setClickState] = useState<boolean>(false);

  function handleClick(): any {
    setClickState(!clickState);
  }

  return (
    <Fragment>
      <header>
        <Title />
        <Subtitle text="Here is a subtitle" />
      </header>
      <p>The button is{!clickState && " not"} clickified!</p>
      <Button handleClick={handleClick} />
    </Fragment>
  );
};

export default App;