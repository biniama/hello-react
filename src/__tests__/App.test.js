import React from "react";
import { create, act } from "react-test-renderer";
import App from "../App";

describe("App component", () => {
  it("matches App snapshot", () => {
    const component = create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches App snapshot when button is clicked", () => {
    const component = create(<App />);
    const root = component.root;
    const button = root.findByType("button");
    act(() => button.props.onClick());
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches App snapshot when button is clicked twice", () => {
    const component = create(<App />);
    const root = component.root;
    const button = root.findByType("button");
    act(() => button.props.onClick());
    act(() => button.props.onClick());
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
