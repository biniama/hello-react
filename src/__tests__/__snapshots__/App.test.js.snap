// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches App snapshot 1`] = `
Array [
  <header>
    <h1>
      Welcome to React!!!
    </h1>
    <h2>
      HERE IS A SUBTITLE
    </h2>
  </header>,
  <p>
    The button is
     not
     clickified!
  </p>,
  <button
    onClick={[Function]}
  >
    Click me!!!
  </button>,
]
`;

exports[`App component matches App snapshot when button is clicked 1`] = `
Array [
  <header>
    <h1>
      Welcome to React!!!
    </h1>
    <h2>
      HERE IS A SUBTITLE
    </h2>
  </header>,
  <p>
    The button is
     clickified!
  </p>,
  <button
    onClick={[Function]}
  >
    Click me!!!
  </button>,
]
`;

exports[`App component matches App snapshot when button is clicked twice 1`] = `
Array [
  <header>
    <h1>
      Welcome to React!!!
    </h1>
    <h2>
      HERE IS A SUBTITLE
    </h2>
  </header>,
  <p>
    The button is
     not
     clickified!
  </p>,
  <button
    onClick={[Function]}
  >
    Click me!!!
  </button>,
]
`;
